### ISSUE TITLE
Mac/Safari - [AU] Attachment - "WebKitBlobResource error" console error when sending a PDF attachment

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#54640]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.79-2
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/5386106&group_by=cases:section_id&group_id=292107&group_order=asc
**Issue reported by:** Applause Internal Team
**Device used:** macOS Sequoia 15.0 - Safari
**App Component:** Other

## Action Performed:
1. Navigate to https://staging.new.expensify.com/ and open Safari Web Inspector
2. Log in with a new Gmail account
3. Open any 1:1 DM
4. Send any PDF attachment

## Expected Result:
Console error shouldn't show.

## Actual Result:
"WebKitBlobResource error" console error when sending a PDF attachment to a 1:1 DM.

## Workaround:
Unknown

## Platforms:
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos
[Bug6703307_1735366281759!Console.txt](https://github.com/user-attachments/files/18266168/Bug6703307_1735366281759.Console.txt)

https://github.com/user-attachments/assets/dd64cf4b-d85a-47b2-a083-14243ee65ce5

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by kpadmanabhan** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Uploading of attachment in 1:1 DM shows console error

### What is the root cause of that problem?
Though on an initial look, this may point fingers at Safari unable to handle pdf blobs, it is not a Safari alone issue. As against what's tagged in the title of this bug, this occurs on other browsers as well.
![image](https://github.com/user-attachments/assets/c80f2153-95b2-4e65-9834-66c1b8ff5249)
As Blob URLs are temporary in memory references to data from browser memory, a GET call to this URL fails to get it, because the `encryptedAuthToken` is invalid.

### What changes do you think we should make in order to solve the problem?
If we make a call to the blob URL without the encryptedAuthToken, the call succeeds.

The attachment is added without `encryptedAuthToken` in 
https://github.com/Expensify/App/blob/main/src/pages/home/report/ReportActionCompose/ReportActionCompose.tsx#L262
However a further call to get attachment is made using `encryptedAuthToken`, which results in display of error message in console.

Remove `encryptedAuthToken` parameter if reportAction is attachment without any comment.
 
### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?

N.A.

### What alternative solutions did you explore? (Optional)
N.A.

**Comment by RachCHopkins** [comment]
I think this is related to last week's fire. @IuliiaHerets can we please get a retest?

**Comment by IuliiaHerets** [comment]
@RachCHopkins Tester still can reproduce this issue in the latest build v9.0.80-1

https://github.com/user-attachments/assets/2778f301-be91-43ef-ac41-0b0f4d763688

**Comment by RachCHopkins** [comment]
I can't repro this at all.

**Comment by RachCHopkins** [comment]
Ok, confirmed Kadie can repro!

**Comment by hungvu193** [comment]
I will review today!

**Comment by hungvu193** [comment]
> , and add `isAuthTokenRequired={!!sourceID}` to add the authToken when required.

@shubham1206agra  Which case do we need this one?

**Comment by hungvu193** [comment]
> > @shubham1206agra Which case do we need this one?
> 
> When the attachment is uploaded, we need to fetch it from a server. That's why we need authToken then.

I can't reproduce it, even when I didn't add `isAuthTokenRequired={!!sourceID}`, there's still no error shown.

**Comment by hungvu193** [comment]
Yep, I can reproduce, I just clarify the part of Shubham proposal above.

**Comment by hungvu193** [comment]
@NJ-2020  `FileUtils.isLocalFile(source) ` will return true when attachment source starts with `/`:
https://github.com/Expensify/App/blob/e933f4ad00c070450f6ae5a643dbe8f7ff6c88a2/src/libs/fileDownload/FileUtils.ts#L292
In this issue, once attachment is loaded from our server, `source` will have the the format ` /chat-attachments/..../.../filename`, if we use `FileUtils.isLocalFile(source)` in our condition, it will always return true, which I think it doesn't correct.

**Comment by hungvu193** [comment]
@shubham1206agra 's [proposal here ](https://github.com/Expensify/App/issues/54640#issuecomment-2564648498)LGTM, we can use `sourceID` to detect when the attachment is loaded from our server, we also used it everywhere inside our `BaseAnchorForAttachmentsOnly` component.

🎀 👀 🎀  C+ reviewed

**Comment by mjasikowski** [comment]
@shubham1206agra's proposal looks good!

**Comment by hungvu193** [comment]
Any ETA for the PR? @shubham1206agra

**Comment by hungvu193** [comment]
# BugZero Checklist:

- [x] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [ ] 1a. Result of the original design (eg. a case wasn't considered)
  - [x] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [ ] 1z. Other:

Where bug was reported:
  - [x] 2a. Reported on production (eg. bug slipped through the normal regression and PR testing process on staging)
  - [ ] 2b. Reported on staging (eg. found during regression or PR testing)
  - [ ] 2d. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [ ] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [x] 3d. QA
  - [ ] 3z. Other:

</details>

- [x] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment: https://github.com/Expensify/App/pull/10144/files#r1921945967

- [x] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion: N/A

- [x] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.


## Regression Test Proposal

### Test:


1. Open App.
2. Open any 1:1 DM.
3. Send any PDF attachments.
4. Verify that there's no error in the console.

Do we agree 👍 or 👎

**Comment by garrettmknight** [comment]
$250 approved for @hungvu193

**Comment by RachCHopkins** [comment]
Payment Summary:

- Contributor: @shubham1206agra to be paid $250 via Upwork
- Contributor+: @hungvu193 paid $250 via NewDot Manual Request

Upwork job [here](https://www.upwork.com/jobs/~021876132827778405205)
